src/gadgets/bits2num.rs

```diff
@@ -0,0 +1,270 @@
+use ff::{Field, PrimeField};
+use halo2_proofs::circuit::{AssignedCell, Layouter, Value};
+use halo2_proofs::plonk::{Advice, Column, ConstraintSystem, Constraints, Error, Selector};
+use halo2_proofs::poly::Rotation;
+use std::marker::PhantomData;
+
+pub(crate) trait Bits2NumInstruction<F: Field> {
+
+    /// Convert the bits in big endian order to a number
+    fn convert_be(
+        &self,
+        layouter: &mut impl Layouter<F>,
+        bits: Vec<AssignedCell<F, F>>,
+    ) -> Result<AssignedCell<F, F>, Error>;
+
+    /// Convert the bits in little endian order to a number
+    fn convert_le(
+        &self,
+        layouter: &mut impl Layouter<F>,
+        bits: Vec<AssignedCell<F, F>>,
+    ) -> Result<AssignedCell<F, F>, Error>;
+}
+
+
+#[derive(Debug, Clone)]
+pub(crate) struct Bits2NumChipConfig {
+    pub(crate) selector: Selector,
+    pub(crate) input: Column<Advice>,
+    pub(crate) output: Column<Advice>,
+}
+
+pub(crate) struct Bits2NumChip<F: Field> {
```

Comment:
I started adding comprehensive documentation to all chips ([example](https://github.com/zkp-gravity/0g-halo2/blob/main/src/gadgets/bloom_filter/byte_selector.rs#L42-L64)), would be nice to add here as well! I highly recommend [this tool](https://www.tablesgenerator.com/markdown_tables) to generate the tables.

Reply:
For example, it's important to note that the inputs are assumed to be range-checked already (which is actually not the case right now, but will be once the thermometer encoding is done).